### PR TITLE
Oshadura/gh 142 neon floatloads

### PR DESCRIPTION
**Pull request by oshadura**
Passing simple load/store tests for floats

### CODE REVIEW

File: Vc/detail/neon.h [hunk 1]
```diff
@@ -129,7 +281,7 @@ struct neon_datapar_impl : public generic_datapar_impl<neon_datapar_impl> {
     {
         return detail::load16(mem, f);
     }
-  
+
```
Comment (mattkretz) [inline Vc/detail/neon.h]: trailing space, please remove

File: Vc/detail/neon.h [hunk 2]
```diff
@@ -139,16 +291,16 @@ struct neon_datapar_impl : public generic_datapar_impl<neon_datapar_impl> {
         // alignment F doesn't matter
         execute_n_times<size<T>()>([&](auto i) { mem[i] = v.m(i); });
     }
- 
+
```
Comment (mattkretz) [inline Vc/detail/neon.h]: trailing space

File: Vc/detail/neon.h [hunk 3]
```diff
@@ -168,7 +320,7 @@ struct neon_datapar_impl : public generic_datapar_impl<neon_datapar_impl> {
         execute_n_times<size<T>()>([&](auto i) { mem[i] = static_cast<U>(v[i]); });
 #endif
     }
- 
+
```
Comment (mattkretz) [inline Vc/detail/neon.h]: ditto

File: Vc/detail/neon.h [hunk 4]
```diff
@@ -181,7 +333,7 @@ struct neon_datapar_impl : public generic_datapar_impl<neon_datapar_impl> {
         execute_n_times<size<T>()>([&](auto i) { mem[i] = static_cast<U>(v[i]); });
 #endif
     }
-
+
```
Comment (mattkretz) [inline Vc/detail/neon.h]: ... and more, please just search for / $/ and remove all trailing spaces

File: Vc/detail/storage.h [hunk 5]
```diff
@@ -618,7 +622,343 @@ inline std::basic_ostream<CharT> &operator<<(std::basic_ostream<CharT> & s,
 }
 
 //}}}1
-#endif  // Vc_HAVE_SSE
+#elif defined Vc_HAVE_NEON
```
Comment (mattkretz) [inline Vc/detail/storage.h]: OK, we talked about this. We need to get rid of this copy ASAP. Not for this PR, but that should probably be the next PR.

File: Vc/detail/neon.h (hunk 1 above)
Comment (oshadura) [inline Vc/detail/neon.h]: fixed

File: Vc/detail/neon.h (hunk 2 above)
Comment (oshadura) [inline Vc/detail/neon.h]: fixed

File: Vc/detail/neon.h (hunk 3 above)
Comment (oshadura) [inline Vc/detail/neon.h]: fixed

File: Vc/detail/neon.h (hunk 4 above)
Comment (oshadura) [inline Vc/detail/neon.h]: uh, you are right, sorry - fixed

**Pull request merged**